src/ptp.rs

```diff
@@ -110,9 +112,47 @@ pub trait PtpCommandCode: Sized + Copy {
             *a as u16
         }
     }
+
+    fn enum_name(&self) -> String;
 }
 
-impl PtpCommandCode for StandardCommandCode {}
+impl PtpCommandCode for StandardCommandCode {
+    fn enum_name(&self) -> String {
+        match *self {
```

Comment:
You should add a `use StandardCommandCode::*` here and move all the `StandardCommandCode::` prefixes below.

Also, does `format!("{:?}", self)` not produce the name of the enum? Just curious, might save hardcoding all this.